### PR TITLE
brand and collection trigger 400; videoUrl parsing

### PR DESCRIPTION
**Pull request by UladBohdan**
* Backend will now return `400 Bad Request` on requests containing `brand` or `collection` fields.
* `videoUrl` parsing

### CODE REVIEW

**Comment by Drapegnik** [comment]
[500 on article update](https://trello.com/c/bOm95Qw0/235-500-on-article-update)

**Comment by UladBohdan** [comment]
`videoUrl/videoId` changes are in sync with https://github.com/babajka/babajka-frontend/pull/36

**Comment by UladBohdan** [comment]
[Video article page](https://trello.com/c/iSx79dwg/137-video-article-page)

File: src/api/article/controller.tests.js [hunk 1]
```diff
@@ -420,6 +488,34 @@ describe('Articles Bundled API', () => {
         expect(res.body.locales.be.slug).to.equal('be-slug');
       }));
 
+  it('should update article video reference', () =>
+    request
+      .put('/api/articles/be-slug')
+      .set('Cookie', sessionCookie)
+      .send({
+        videoUrl: 'https://www.youtube.com/watch?v=ABCABCABCAB',
```
Comment (Drapegnik) [inline src/api/article/controller.tests.js]: mb better extract that id to variable?

File: src/utils/validation.js [hunk 2]
```diff
@@ -26,15 +33,25 @@ const createArticleValidator = ({ body }, res, next) => {
...
+    errors.video = 'errors.forbiddenForTypeText';
+  }
+
   return next(!isEmpty(errors) && new ValidationError(errors));
 };
 
 const updateArticleValidator = ({ body }, res, next) => {
   const errors = {};
 
+  ['brand', 'collection'].forEach(field => {
```
Comment (Drapegnik) [inline src/utils/validation.js]: please extract it in function & remove copy-pasting above

File: src/api/article/controller.js [hunk 3]
```diff
@@ -165,6 +168,13 @@ export const update = async ({ params: { slugOrId }, body, user }, res, next) =>
       article[key] = value;
     });
 
+    if (article.type === 'video' && body.videoUrl) {
+      article.video = parseVideoUrl(body.videoUrl);
+    }
+    if (article.type === 'text') {
+      article.video = undefined;
```
[the remark this answers is not in the document]
Comment (UladBohdan) [reply src/api/article/controller.js]: this is to pass validation checking that `type=text` articles are free of any video information. Not necessary, actually, but helps keep db clean.

File: src/api/article/controller.tests.js (hunk 1 above)
Comment (UladBohdan) [reply src/api/article/controller.tests.js]: extracted.

File: src/utils/validation.js (hunk 2 above)
Comment (UladBohdan) [reply src/utils/validation.js]: done.

**Pull request merged**